src/renderers/native/ReactNativeFiber.js

```diff
@@ -0,0 +1,378 @@
+
+    if (__DEV__) {
+      for (let key in viewConfig.validAttributes) {
+        if (props.hasOwnProperty(key)) {
+          deepFreezeAndThrowOnMutationInDev(props[key]);
+        }
+      }
+    }
+
+    const updatePayload = ReactNativeAttributePayload.create(
+      props,
+      viewConfig.validAttributes
+    );
+
+    UIManager.createView(
+      tag, // reactTag
+      viewConfig.uiViewClassName, // viewName
+      rootContainerInstance, // rootTag
+      (updatePayload : any), // props
+    );
+
+    const component = new NativeHostComponent(tag);
+
+    precacheFiberNode(internalInstanceHandle, tag);
+
+    return component;
+  },
+
+  createTextInstance(
+    text : string,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : TextInstance {
+    const tag = ReactNativeTagHandles.allocateTag();
+
+    UIManager.createView(
+      tag, // reactTag
+      'RCTRawText', // viewName
+      rootContainerInstance, // rootTag
+      {text: text} // props
+    );
+
+    precacheFiberNode(internalInstanceHandle, tag);
+
+    return tag;
+  },
+
+  finalizeInitialChildren(
+    parentInstance : Instance,
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+  ) : void {
+    // Map from child objects to native tags.
+    // Either way we need to pass a copy of the Array to prevent it from being frozen.
+    const nativeTags = parentInstance._children.map(
+      (child) => typeof child === 'number'
+        ? child // Leaf node (eg text)
+        : child._nativeTag
+    );
+
+    UIManager.setChildren(
+      parentInstance._nativeTag, // containerTag
+      nativeTags // reactTags
+    );
+  },
+
+  getChildHostContext(
+    parentHostContext : string | null,
+    type : string
+  ) {
+    return parentHostContext;
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    const children = (parentInstance : any)._children;
+    const beforeChildIndex = children.indexOf(beforeChild);
+    const index = children.indexOf(child);
+
+    // Move existing child or add new child?
+    if (index >= 0) {
+      children.splice(index, 1);
+      children.splice(beforeChildIndex, 0, child);
+
+      UIManager.manageChildren(
+        (parentInstance : any)._nativeTag, // containerID
+        [index], // moveFromIndices
+        [beforeChildIndex], // moveToIndices
+        [], // addChildReactTags
+        [], // addAtIndices
+        [], // removeAtIndices
+      );
+    } else {
+      children.splice(beforeChildIndex, 0, child);
+
+      UIManager.manageChildren(
+        (parentInstance : any)._nativeTag, // containerID
+        [], // moveFromIndices
+        [], // moveToIndices
+        [(child : any)._nativeTag], // addChildReactTags
+        [beforeChildIndex], // addAtIndices
+        [], // removeAtIndices
+      );
+    }
+  },
+
+  prepareForCommit() : void {
+    // Noop?
```

Comment:
Yes, this corresponds to the wrappers in ReactNativeReconcileTransaction and should be a noop.